File: Archive/mad/prg/Activities/Practica5.1/src/Tree.java
// Requires the Java util Scanner class
import java.util.Scanner;

/* The class Tree generates a tree based on the input and prints the total height of it
* @author Daniel "BlueHarrier" Píriz
* @version 1.0.2
* @since 5/11/2022
*/

public class Tree {
	/* Main function. Counts and prints the height of a given number of trees.
	* @param String[] Arguments of the program
	*/
	public static void main(String args[]){
		// Initialize input
		Scanner in  = new Scanner(System.in);
		
		// Create an entire tree based on the input and the number of cases
		int cases = in.nextInt();
		for (int i = 0; i < cases; i++){
			Branch root = new Branch(in);
			System.out.println(Integer.toString(root.getHeight()));
		}
	}
}

File: Archive/mad/prg/Activities/Practica5.1/cpp/src/Tree.cpp
// IOStream library
#include <iostream>
using namespace std;

// Branch class
class Branch{
    public : int calculateHeight(){
        // Initialize the branches and current height
        int branches;
        cin >> branches;
        int height = 0;

        // Execute each branch
        Branch* branch = new Branch();
        for (int i = 0; i < branches; i++){
            int branchHeight = branch->calculateHeight();
            if (branchHeight > height){
                height = branchHeight;
            }
        }
        delete branch;

        // Add and return height
        height++;
        return height;
    }
};

// Main function
int main() {
    // Pick up the number of cases
    int cases;
    cin >> cases;

    // Execute for each tree
    Branch* root = new Branch();
    for (int i = 0; i < cases; i++){
        cout << root->calculateHeight() << "\n";
    }
    delete root;
}